### ISSUE TITLE
BASIC tier unavailable

### ISSUE THREAD

**Issue by AlejandroCasa** [terraform-google-modules/terraform-google-memorystore#97]
### TL;DR

In case of declaring the tier as BASIC, the Terraform fails because the variables replica_count and read_replicas_mode are not expected.

### Expected behavior

module.memorystore["production"].google_redis_instance.default will be created
  + resource "google_redis_instance" "default" {
      + alternative_location_id  = (known after apply)
      + auth_enabled             = false
      + auth_string              = (sensitive value)
      + authorized_network       = "projects/xxxxxxxxxxx/global/networks/yyyyyy"
      + connect_mode             = "PRIVATE_SERVICE_ACCESS"
      + create_time              = (known after apply)
      + current_location_id      = (known after apply)
      + host                     = (known after apply)
      + id                       = (known after apply)
      + location_id              = (known after apply)
      + memory_size_gb           = 1
      + name                     = "my-demo-tf"
      + persistence_iam_identity = (known after apply)
      + port                     = (known after apply)
      + project                  = "xxxx"
      + redis_version            = "REDIS_6_X"
      + region                   = "europe-west1"
      + reserved_ip_range        = (known after apply)
      + server_ca_certs          = (known after apply)
      + tier                     = "BASIC"
      + transit_encryption_mode  = "SERVER_AUTHENTICATION"
    }

### Observed behavior

Terraform v1.2.3
on linux_amd64
Initializing plugins and modules...
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/memorystore/main.tf line 23, in resource "google_redis_instance" "default":
│   23:   replica_count      = var.tier == "STANDARD_HA" ? var.replica_count : null
│ 
│ An argument named "replica_count" is not expected here.
╵
╷
│ Error: Unsupported argument
│ 
│   on .terraform/modules/memorystore/main.tf line 24, in resource "google_redis_instance" "default":
│   24:   read_replicas_mode = var.tier == "STANDARD_HA" ? var.read_replicas_mode : null
│ 
│ An argument named "read_replicas_mode" is not expected here.
╵
Operation failed: failed running terraform plan (exit 1)

### Terraform Configuration

```hcl
module "memorystore" {
  for_each = local.wallace_backend_configurations

  source  = "terraform-google-modules/memorystore/google"
  version = "4.4.1"

  name               = "my-demo-tf"
  project            = module.project.project_id[each.key]
  region             = var.project_configuration["default"]["region"]
  
  tier               = "BASIC"
  redis_version      = "REDIS_6_X"
  memory_size_gb     = 1

  authorized_network = "projects/XXXXXXXXXX/global/networks/YYYYYYYY"
  connect_mode       = "PRIVATE_SERVICE_ACCESS"
}
```

### Terraform Version

```sh
v1.2.3
```

### Additional information

With module version fixed as 4.0.0, it works perfectly
